crates/bvs-vault-cw20-tokenized/src/contract.rs

```diff
@@ -0,0 +1,707 @@
+        .add_attribute("router", router)
+        .add_attribute("operator", operator)
+        .add_attribute("staking_cw20_contract", cw20_contract);
+
+    Ok(response)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: CombinedExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        CombinedExecuteMsg::WithdrawTo(msg) => {
+            // This is the only execute msg that is not passed to the base contract
+            // because it is a custom logic for this vault contract
+            vault_execute::withdraw_to(deps, env, info, msg)
+        }
+        CombinedExecuteMsg::DepositFor(msg) => {
+            // This is the only execute msg that is not passed to the base contract
+            // because it is a custom logic for this vault contract
+            vault_execute::deposit_for(deps, env, info, msg)
+        }
+        CombinedExecuteMsg::QueueWithdrawalTo(msg) => {
+            // This is the only execute msg that is not passed to the base contract
+            // because it is a custom logic for this vault contract
+            vault_execute::queue_withdrawal_to(deps, env, info, msg)
+        }
+        CombinedExecuteMsg::RedeemWithdrawalTo(msg) => {
+            // This is the only execute msg that is not passed to the base contract
+            // because it is a custom logic for this vault contract
+            vault_execute::redeem_withdrawal_to(deps, env, info, msg)
+        }
+        _ => {
+            // cw20 compliant messages are passed to the `cw20-base` contract.
+            // Except for the `Burn` and `BurnFrom` messages.
+            receipt_cw20_execute::execute_base(deps, env, info, msg).map_err(Into::into)
+        }
+    }
+}
+
+/// cw20 compliant messages are passed to the `cw20-base` contract.
+/// Except for the `Mint`, `Burn` and `BurnFrom` messages.
+/// The only time receipt token total supply should be changed is through staking and unstaking
+/// More precisely, only through - deposit_for and withdraw_to and redeem_withdrawal_to
+mod receipt_cw20_execute {
+    use cosmwasm_std::{Addr, StdError, StdResult, Uint128};
+    use cosmwasm_std::{DepsMut, Env, MessageInfo, Response};
+
+    use cw20_base::contract::execute_send;
+    use cw20_base::contract::execute_transfer;
+
+    use cw20_base::allowances::execute_decrease_allowance;
+    use cw20_base::allowances::execute_increase_allowance;
+    use cw20_base::allowances::execute_send_from;
+    use cw20_base::allowances::execute_transfer_from;
+
+    use cw20_base::state::{BALANCES as RECEIPT_TOKEN_BALANCES, TOKEN_INFO as RECEIPT_TOKEN_INFO};
+
+    use crate::msg::ExecuteMsg as CombinedExecuteMsg;
+
+    /// This mint function is almost identical to the base cw20 contract's mint function
+    /// down to the variables and logic.
+    /// Except that it does not require the caller to be the minter.
+    pub fn mint(
```

Comment:
```suggestion
    pub fn mint_internal(
```

Also call it mint_internal?